src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -261,27 +280,38 @@ module.exports = function<T, P, I>(config : HostConfig<T, P, I>) : Reconciler {
   return {
 
     mountContainer(element : ReactElement<any>, containerInfo : ?Object) : OpaqueNode {
-      const container = ReactFiber.createHostContainerFiber(containerInfo, LowPriority);
-      container.alternate = container;
+      const root = createFiberRoot(containerInfo);
+      const container = root.current;
+      // TODO: Use pending work/state instead of props.
       container.pendingProps = element;
+      container.pendingWorkPriority = LowPriority;
 
-      scheduleLowPriWork(container);
+      scheduleLowPriWork(root);
 
+      // It may seem strange that we don't return the root here, but that will
+      // allow us to have containers that are in the middle of the tree instead
+      // of being roots.
```

Comment:
It allows `mountContainer` or at least `updateContainer` to remain agnostic to whether they're updating something that is a root or part of the tree.

Not sure if that is the API we're going for or maybe an explicit API like `renderSubtreeIntoContainer`. If it is explicit, then I guess these could be different functions anyway.

Meh.